tests/backup/backup_helper.go

```diff
@@ -0,0 +1,1234 @@
+			defer GinkgoRecover()
+			defer wg.Done()
+			err := backup.AddUser(userName, firstName, lastName, email, password)
+			Inst().Dash.VerifyFatal(err, nil, fmt.Sprintf("Creating user - %s", userName))
+			users = append(users, userName)
+		}(userName, firstName, lastName, email)
+	}
+	wg.Wait()
+	return users
+}
+
+func DeleteCloudAccounts(backupLocationMap map[string]string, credName string, cloudCredUID string, ctx context.Context) {
+	if len(backupLocationMap) != 0 {
+		for backupLocationUID, bkpLocationName := range backupLocationMap {
+			err := DeleteBackupLocation(bkpLocationName, backupLocationUID, orgID)
+			Inst().Dash.VerifySafely(err, nil, fmt.Sprintf("Deleting backup location %s", bkpLocationName))
+		}
+		time.Sleep(time.Minute * 3)
+		err := DeleteCloudCredential(credName, orgID, cloudCredUID)
+		Inst().Dash.VerifySafely(err, nil, fmt.Sprintf("Deleting cloud cred %s", credName))
+	}
+	err := DeleteCluster(SourceClusterName, orgID, ctx)
+	Inst().Dash.VerifySafely(err, nil, fmt.Sprintf("Deleting cluster %s", SourceClusterName))
+	err = DeleteCluster(destinationClusterName, orgID, ctx)
+	Inst().Dash.VerifySafely(err, nil, fmt.Sprintf("Deleting cluster %s", destinationClusterName))
+}
+
+// AddRoleAndAccessToUsers assigns role and access level to the users
+// AddRoleAndAccessToUsers return map whose key is userRoleAccess and value is backup for each user
+func AddRoleAndAccessToUsers(users []string, backupNames []string) (map[userRoleAccess]string, error) {
+	var access BackupAccess
+	var role backup.PxBackupRole
+	roleAccessUserBackupContext := make(map[userRoleAccess]string)
+	ctx, err := backup.GetAdminCtxFromSecret()
+	if err != nil {
+		return nil, err
+	}
+	for i := 0; i < len(users); i++ {
+		userIndex := i % 9
+		switch userIndex {
+		case 0:
+			access = ViewOnlyAccess
+			role = backup.ApplicationOwner
+		case 1:
+			access = RestoreAccess
+			role = backup.ApplicationOwner
+		case 2:
+			access = FullAccess
+			role = backup.ApplicationOwner
+		case 3:
+			access = ViewOnlyAccess
+			role = backup.ApplicationUser
+		case 4:
+			access = RestoreAccess
+			role = backup.ApplicationUser
+		case 5:
+			access = FullAccess
+			role = backup.ApplicationUser
+		case 6:
+			access = ViewOnlyAccess
+			role = backup.InfrastructureOwner
+		case 7:
+			access = RestoreAccess
+			role = backup.InfrastructureOwner
+		case 8:
+			access = FullAccess
+			role = backup.InfrastructureOwner
+		default:
+			access = ViewOnlyAccess
+			role = backup.ApplicationOwner
+		}
+		ctxNonAdmin, err := backup.GetNonAdminCtx(users[i], "Password1")
+		if err != nil {
+			return nil, err
+		}
+		userRoleAccessContext := userRoleAccess{users[i], role, access, ctxNonAdmin}
+		roleAccessUserBackupContext[userRoleAccessContext] = backupNames[i]
+		err = backup.AddRoleToUser(users[i], role, "Adding role to user")
+		if err != nil {
+			err = fmt.Errorf("Failed to add role %s to user %s with err %v", role, users[i], err)
```

Comment:
Make all letter lower case in error message

Reply:
done